src/System.Runtime.Extensions/src/System/IO/Path.cs

```diff
@@ -356,58 +357,58 @@ private static string CombineNoChecks(string path1, string path2, string path3)
                 'q', 'r', 's', 't', 'u', 'v', 'w', 'x',
                 'y', 'z', '0', '1', '2', '3', '4', '5'};
 
-        private static string ToBase32StringSuitableForDirName(byte[] buff)
+        private static unsafe void GetBase32CharsSuitableForDirName(byte* bytes, int byteCount, char* chars, int charCount)
```

Comment:
Nit: Given the change in implementation, I'm not sure this name is still the best one.  How about something like Populate83FileNameFromRandomBytes, or something like that.  The ```pRandomFileName[8] = '.'``` could then be moved into the function, instead of the code currently setting the 8th char.